dvc/fs/http.py

```diff
@@ -190,6 +190,15 @@ def _upload(
                 desc=name or to_info.url,
             )
 
+    def open(self, path_info, mode: str = "r", encoding: str = None, **kwargs):
+        return super().open(
+            path_info,
+            mode,
+            encoding,
+            auth=self._auth_method(path_info),
+            **kwargs,
+        )
+
```

Comment:
Looks like our httpfs is still using an old `_generate_download_url` approach, which we are getting rid of in favor of just using fsspec-compatible filesystems. So to be correct, we just need to move stuff from `BaseFileSystem.open` into `HTTPFileSystem.open` here, so we won't need to do this confusing `super()` call at all. @simonlsk Could you adjust that, please?